BranchMetrics/ios-spm -> BranchMetrics/ios-branch-sdk-spm

jdee opened the pull request:
The package(s) being submitted are:

* [Branch](https://github.com/BranchMetrics/ios-branch-sdk-spm)

Note: We renamed this repo. Both names work because GitHub redirects. In some views, only the bare repo name appears (ios-spm). This repo name is less confusing. See https://help.branch.io/developers-hub/docs/ios-basic-integration#via-swift-package-manager for more information.

## Checklist

I have either:

* [x ] Run `swift ./validate.swift`.

Or, checked that:

* [ ] The package repositories are publicly accessible.
* [ ] The packages all contain a `Package.swift` file in the root folder.
* [ ] The packages are written in Swift 5.0 or later.
* [ ] The packages all contain at least one product (either library or executable), and at least one product is usable in other Swift apps.
* [ ] The packages all have at least one release tagged as a [semantic version](https://semver.org/).
* [ ] The packages all output valid JSON from `swift package dump-package` with the latest Swift toolchain.
* [ ] The package URLs are all fully specified including the protocol (usually `https`) and the `.git` extension.
* [ ] The packages all compile without errors.
* [ ] The package list JSON file is sorted alphabetically.

jdee commented:
We need a means to remove the ios-spm repo from this list. Leaving it here will be unhelpful.

daveverwer commented:
Thanks for this Jimmy!

If I understand correctly, you'd like to replace `BranchMetrics/ios-spm ` with `BranchMetrics/ios-branch-sdk-spm.git`. Is that correct?

If so, just amend this PR to remove the line with `BranchMetrics/ios-spm` in it. The index will take that as a deletion when it sees this new JSON file.

Thanks!
Dave

jdee commented:
@daveverwer Thank you. Validation originally failed when I removed the old repo. It was because `validate.swift` made a whitespace change I needed to commit. Updated here.

daveverwer commented:
Yea. I wish whitespace didn't trip up our validation.

Thanks for the amendment!
